pulsar-transformations/src/main/resources/config-schema.yaml

```diff
@@ -141,6 +142,57 @@ components:
               description: The delimiter to use when concatenating the field names.
               default: _
         - "$ref": "#/components/schemas/Part"
+    ComputeFields:
+      allOf:
+        - "$ref": "#/components/schemas/Step"
+        - type: object
+          description: Adds or updates fields dynamically based on an expression (Currently only AVRO is supported).
+          properties:
+            type:
+              type: string
+              enum:
+                - compute-fields
+            fields:
+              type: [ array, 'null' ]
+              items:
+                type:
+                  - object
+                minLength: 1
+                properties:
+                  name:
+                    type: string
+                    description: The name of the field to compute. If the field name already exist, it will be overwritten.
+                  expression:
+                    type: string
+                    description: An EL-like expression to compute the field value. It can refer to other attributes in the record.
+                  type:
+                    type: string
```

Comment:
do we allow null ?

Reply:
not really, this should be [key, value] or not provided. null will override the default value which is value

Reply:
hmm, I wonder what is the use case? Will leave it out for now.

Reply:
So I'll make the field optional (so a union of nullable, type) as per Enrico's. So the user still has to provide an explicit type, but if the fields evaluates to null, it will not break their schema.